File: csharp/bird-watcher/BirdWatcher.cs
class BirdCount
{
    private int[] birdsPerDay;

    public BirdCount(int[] birdsPerDay)
    {
        this.birdsPerDay = birdsPerDay;
    }

    public static int[] LastWeek()
        => [0, 2, 5, 3, 7, 8, 4];

    public int Today()
        => this.birdsPerDay.Last();

    public void IncrementTodaysCount()
    {
        var temp = this.birdsPerDay.Last();
        temp++;
        this.birdsPerDay.SetValue(temp, this.birdsPerDay.Length-1);
    }

    public bool HasDayWithoutBirds()
        => this.birdsPerDay.Any(i => i == 0);

    public int CountForFirstDays(int numberOfDays)
    {
        int s = 0;
        for(int i = 0; i < numberOfDays; i++)
        {
            s += this.birdsPerDay[i];
        }
        return s;
    }

    public int BusyDays()
    {
        int busyDays = 0;
        foreach (int day in this.birdsPerDay)
        {
            if (day >= 5)
            {
                busyDays++;
            }
        }
        return busyDays;
    }
}

File: java/bird-watcher/src/main/java/BirdWatcher.java

class BirdWatcher {
    private final int[] birdsPerDay;

    public BirdWatcher(int[] birdsPerDay) {
        this.birdsPerDay = birdsPerDay.clone();
    }

    public int[] getLastWeek() {
        int[] lastWeek = new int[] { 0, 2, 5, 3, 7, 8, 4};
        return lastWeek;
    }

    public int getToday() {
        return this.birdsPerDay[this.birdsPerDay.length-1];
    }

    public void incrementTodaysCount() {
        this.birdsPerDay[this.birdsPerDay.length-1]++;
    }

    public boolean hasDayWithoutBirds() {
        for (int birds : birdsPerDay) {
            if (birds == 0) {
                return true;
            }
        }
        return false;
    }

    public int getCountForFirstDays(int numberOfDays) {
        int sum = 0;
        int realNumberOfDays = numberOfDays > birdsPerDay.length 
            ? birdsPerDay.length 
            : numberOfDays;
        for (int i = 0; i < realNumberOfDays; i++) {
            sum += this.birdsPerDay[i];
        }
        return sum;
    }

    public int getBusyDays() {
        int busyDays = 0;
        for (int birds : birdsPerDay) {
            if (birds >= 5) {
                busyDays++;
            }
        }
        return busyDays;
    }
}
